### PR TITLE
Feature/hiv indicators

### LINKED ISSUE #1685: Indicators discussion
## Description

Epic for organising indicators work for `v3.2.0` release.

### Context

HIV indicators.

## Design discussion

As the desktop indicators implementation is still awaiting some feedback (in particular re. migration and sync) there are still some unknowns which make it hard to make informed back-end design decisions. If others agree, it seems like a good idea to focus discussion on front-end for now, just so that we can keep mobile and desktop development on indicators going in parallel.

Initial thoughts on general design/implementation process:

### Front-end

- [x] UI design.
   - [x] Wireframe e.g. prototypes.
   - [x] Evaluate designs (time constraints, performance, UX etc.).
   - [x] Wireframe final modal/page designs.
- [x] Components
  - [x] Decide what existing/new components are needed to implement chosen design.
  - [x] Implement any new components.
- [x] Navigation
  - [x] Decide navigation logic between any new pages, modals. 
  - [x] Implement navigation logic.

### Back-end

- [x] Database
  - [x] Decide what data is needed from desktop.
  - [x] Design indicators schema.
  - [x] Implement new tables, fields.
- [x] Sync
  - [x] Implement incoming sync logic.
  - [x] Implement version migration logic (including messages?).
  - [x] Implement outgoing sync logic.
- [x] Logic
  - [x] Implement indicators toggle.
  - [x] Implement indicators dropdown.
  - [x] Implement indicators dynamic columns.
  - [x] Implement indicator values rendering to data table.
  - [x] Implement indicator values cell editing.

### Tab (items)

prototype-tab-items

### Tab (indicators, no selection)

prototype-tab-indicators-no-select

### Tab (indicators, selection, not expanded)

prototype-tab-indicators-select-no-expand

### Tab (indicators, selection, expanded)

prototype-tab-indicators-select-expand

### Page (no selection). 

prototype-page-indicators-no-select

### Page (selection).

prototype-page-indicators-select

### Stepper.

prototype-stepper

## Issues.

- #1823.
- #1828.
- #1829.
- #1830.
- #1831.
- #1842.
- #1852.
- #1877.
- #1901.
- #1902.
- #1903.
- #1906.
- #1913.
- #1915.
- #1925.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #1685.

## Change summary

Indicators feature for v3.2.0.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

TODO.

### Related areas to think about

N/A.

### CODE REVIEW

File: src/pages/SupplierRequisitionPage.js [hunk 1]
```diff
@@ -238,25 +260,44 @@ const SupplierRequisition = ({
...
-  }, [showAll, isFinalised]);
+  }, [showIndicators, selectedIndicator, showAll, isFinalised]);
 
   const {
     pageTopSectionContainer,
     pageTopLeftSectionContainer,
     pageTopRightSectionContainer,
   } = globalStyles;
+
+  // TODO: add logic for swapping between items/indicators data.
```
Comment (joshxg) [inline src/pages/SupplierRequisitionPage.js]: TODO's are niiice 🤣

File: src/pages/SupplierRequisitionPage.js [hunk 2]
```diff
@@ -298,6 +339,8 @@ const SupplierRequisition = ({
 };
 
 const mapDispatchToProps = (dispatch, ownProps) => {
+  // TODO: add actions/reducers for indicators toggle.
```
Comment (joshxg) [inline src/pages/SupplierRequisitionPage.js]: shocked

Comment (wlthomson) [reply src/pages/SupplierRequisitionPage.js]: b9cff0eb11329eddf3f89d2f128d7cacfa82fb0e.

File: src/pages/SupplierRequisitionPage.js (hunk 1 above)
Comment (wlthomson) [reply src/pages/SupplierRequisitionPage.js]: dammmn. b9cff0eb11329eddf3f89d2f128d7cacfa82fb0e.

**Pull request merged**